### DISCUSSION TITLE
为什么kfk的副本机制比rabbit的镜像集群好？

### DISCUSSION THREAD

**Discussion by wanli420122** [doocs/advanced-java#181] [Q&A]
如作者描述，rabbitmq镜像集群每个节点都是完整镜像，每个节点需要进行消息同步。kfk采用leader和follow也是需要消息主从同步，我没搞明白这两个之前真正高可用的区别在那里，望高人指点

**Comment by huifer** [comment]
镜像集群每个实例上面都需要存储完整的信息。
在镜像集群中 queue 的meta data (元数据) 都需要发送给集群中的每一个机器, 消息同步占用网络带宽和资源消费可能是一个相对较大的数值.  

副本机制，  数据只有N分之一的数据在一个实例上,  
扩容时的比较 ...
